src/api/spec/models/obs_factory/distribution_spec.rb

```diff
@@ -1,17 +1,229 @@
+    it { expect(distribution.strategy).to                    be_kind_of ObsFactory::DistributionStrategyFactory }
     it { expect(strategy_for('openSUSE:Factory:PowerPC')).to be_kind_of ObsFactory::DistributionStrategyFactoryPPC }
     it { expect(strategy_for('openSUSE:42.3')).to            be_kind_of ObsFactory::DistributionStrategyOpenSUSE }
     it { expect(strategy_for('SUSE:SLE-12-SP1:GA')).to       be_kind_of ObsFactory::DistributionStrategySLE12SP1 }
     it { expect(strategy_for('SUSE:SLE-15:GA')).to           be_kind_of ObsFactory::DistributionStrategySLE15 }
     it { expect(strategy_for('SUSE:SLE-15-SP1:GA')).to       be_kind_of ObsFactory::DistributionStrategySLE15 }
     it { expect(strategy_for('SUSE:SLE-12-SP3:Update:Products:CASP20')).to be_kind_of ObsFactory::DistributionStrategyCasp }
   end
+
+  describe 'self.attributes' do
+    it { expect(ObsFactory::Distribution.attributes.class).to eq(Array) }
+    it { expect(ObsFactory::Distribution.attributes.count).to eq(13) }
+  end
+
+  describe 'self.find' do
+    context 'with a project' do
+      context 'with distribution' do
+        subject { ObsFactory::Distribution.find(project.name) }
+
+        it { expect(subject.class).to eq(ObsFactory::Distribution) }
+      end
+
+      context 'without distribution' do
+        subject { ObsFactory::Distribution.find(other_project.name) }
+
+        it { expect(subject).to be_nil }
+      end
+    end
+
+    context 'with a non-existant project' do
+      subject { ObsFactory::Distribution.find('non-existant') }
+
+      it { expect(subject).to be_nil }
+    end
+  end
+
+  describe '#name' do
+    it { expect(distribution.name).to eq(project.name) }
+  end
+
+  describe '#id' do
+    it { expect(distribution.id).to eq(project.name) }
+  end
+
+  describe '#description' do
+    it { expect(distribution.description).to eq(project.description) }
+  end
+
+  describe '#published_version' do
+    let(:backend_url) { 'http://download.opensuse.org/tumbleweed/repo/oss/media.1/media' }
+    let(:backend_response) do
+      %(openSUSE - openSUSE-20180606-i586-x86_64-Build261.1-Media
+openSUSE-20180606-i586-x86_64-Build261.1
+1
+)
+    end
+
+    before do
+      stub_request(:get, backend_url).and_return(body: backend_response)
+    end
+
+    it { expect(distribution.published_version).to eq('20180606') }
+  end
+
+  describe '#staging_projects' do
+    context 'with staging projects' do
+      let!(:staging_project) { create(:project, name: 'openSUSE:Factory:Staging') }
+      let!(:staging_project_a) { create(:project, name: 'openSUSE:Factory:Staging:A') }
+      let!(:staging_project_b) { create(:project, name: 'openSUSE:Factory:Staging:B') }
+      let!(:staging_project_c) { create(:project, name: 'openSUSE:Factory:Staging:C') }
+      let!(:staging_project_345) { create(:project, name: 'openSUSE:Factory:Staging:345') }
+
+      it { expect(distribution.staging_projects.count).to eq(3) }
```

Comment:
why 3? or better, which 3?

Reply:
I would add a label to this it block that explains what is expected.

And instead of counting you could match the arrays.